### PR TITLE
Opam builder fixes

### PR DESCRIPTION
**Pull request by gasche**

### CODE REVIEW

File: packages/ascii85/ascii85.0.3/opam [hunk 1]
```diff
@@ -13,4 +13,7 @@ build: [
 install: [make "PREFIX=%{prefix}%" "install"]
 remove: [make "PREFIX=%{prefix}%" "remove"]
 available: [ ocaml-version >= "4.01.0" ]
-depends: "ocamlfind"
+depends: [
+  "ocamlfind"
```
Comment (samoht) [inline packages/ascii85/ascii85.0.3/opam]: Should this be a `builld` dependency?

File: packages/fix/fix.20130611/opam [hunk 2]
```diff
@@ -2,5 +2,8 @@ opam-version: "1"
 maintainer: "jonathan.protzenko@gmail.com"
 build: make
 remove: [["ocamlfind" "remove" "fix"]]
-depends: ["ocamlfind"]
+depends: [
+  "ocamlfind"
```
Comment (samoht) [inline packages/fix/fix.20130611/opam]: same here?

**Comment by gasche** [comment]
@samoht I made the ocamlfind dependencies build-only as you recommended.

**Comment by samoht** [comment]
The Travis issues are fixed in opam-depext: https://github.com/ocaml/opam-depext/pull/53

@AltGr is a new release planned out soon?

**Comment by samoht** [comment]
Waiting for https://github.com/ocaml/opam-repository/pull/7170 to restart the jobs.

**Comment by samoht** [comment]
Jobs restarted.

**Comment by lefessan** [comment]
Now that `ocamlfind` installs the `base-bytes` module, extra care must be taken before adding the `{build}` dependency flag, as packages may silently depend on the `Bytes` module installed by `ocamlfind`. Have you checked ?

**Comment by samoht** [comment]
@lefessan that's annoying :-/ any opinion on how we should fix this?

**Comment by dbuenzli** [comment]
This was discussed recently on the platform mailing list. I really think `bytes` should be forked into its own compatibility package like `uchar` and `result` are. We need some uniformity in the handling of these things, the fact that `bytes` is tied to `ocamlfind` makes absolutely no sense.

**Comment by samoht** [comment]
+1 on uniformity. I will be happy to merge things if someone has time to prepare the PRs.

**Comment by lefessan** [comment]
+1

**Comment by gasche** [comment]
We already have a `base-bytes` package to express the dependency on the ocamlfind package `bytes` only. I think you are discussing a problem (or an aspect of it) that is already solved.

There remain reasons to have non-build dependencies on `ocamlfind`: if one uses the `Findlib` library at runtime or if our tool invokes `ocamlfind` through the command-lines. I checked before upgrading these dependencies that it was not the case.

**Comment by samoht** [comment]
ok so merging. Thanks for the clarification @gasche.

**Comment by dbuenzli** [comment]
> We already have a base-bytes package to express the dependency on the ocamlfind package bytes only. I think you are discussing a problem (or an aspect of it) that is already solved.

Right but that's only a tiny aspect. The naming and packaging mess we have is far from being solved. FWIW as I already said elsewhere this `base-bytes` package should be called `bytes`. In general if we want to be able to gradually simplify the system we should

1. Have one OPAM package per `$(lib)` toplevel `ocamlfind` advertises as a base package . These OPAM packages should be called `$(lib)` and *not* `base-$(lib)`. 
2. In general not tuck away compatibility packages in `ocamlfind` as they have nothing to do there.

While 2. is not essential (but I still think that we should treat compatibility packages in a uniform manner), 1 is. 

The reason is that in order to be able to gradually simplify the names we have in the system it is better if we have i) a one-to-one mapping between `ocamlfind` toplevel package names and OPAM package names ii) (If possible) a one-to-one mapping between these package names and the name of the main library archive they expose.

That is unless you like the idea of having to write `package(bytes)` in our `_tags` file but `base-bytes` in your OPAM file.

**Comment by dbuenzli** [comment]
@samoht Since you seem to agree, where should this be filed so that a repo maintainer may eventually consider doing this ? It seems that `opam-repository` issues are mostly ignored.

**Comment by AltGr** [comment]
@dbuenzli, from past experiences, the best way to get through is to have a rewrite script ready, submit a PR and be able to update it when necessary... So even if everybody seems to agree, you'd need to either do it, or convince someone in particular enough so that they will do it.

Part of the `opam-admin.top` helpers was done to that purpose. On the change, I agree that would be much better ; the `provides:` field would also help clean things up (assuming we have good guidelines on its use...), but I'm sorry that's not available yet.

**Comment by dbuenzli** [comment]
> @dbuenzli, from past experiences, the best way to get through is to have a rewrite script ready, submit a PR and be able to update it when necessary... So even if everybody seems to agree, you'd need to either do it, or convince someone in particular enough so that they will do it.

I can assure you that this won't happen from my side.

**Pull request merged**